### PR TITLE
feat: Support for encrypted root disk in node_groups

### PR DESCRIPTION
**Pull request by alzabo**
# PR o'clock

## Description

Add encryption config to node_group launch_group ebs config block.

I'm open to any thoughts about what the keys should be named. I chose to preface with `disk_` for consistency with the other related keys.

### Checklist

- [X] README.md has been updated after any changes to variables and outputs. See https://github.com/terraform-aws-modules/terraform-aws-eks/#doc-generation

### CODE REVIEW

**Comment by jmorgan415** [comment]
+1 for the need for this feature.  For some reason, the managed node group launch templates are lacking quite a few options compared to the self-managed worker launch templates, with disk encryption and metadata options being the most glaring.  I'll submit a PR today for the metadata options as this PR covers the disk encryption quite nicely (I tested successfully on a fork of this repo).

**Comment by dracut5** [comment]
Would be nice to get this functionality, in other case launch template need to be created separately.

File: modules/node_groups/README.md [hunk 1]
```diff
@@ -23,6 +23,8 @@ The role ARN specified in `var.default_iam_role_arn` will be used by default. In
 | capacity\_type | Type of instance capacity to provision. Options are `ON_DEMAND` and `SPOT` | string | Provider default behavior |
 | create_launch_template | Create and use a default launch template | bool |  `false` |
 | desired\_capacity | Desired number of workers | number | `var.workers_group_defaults[asg_desired_capacity]` |
+| disk\_encrypted | Whether the root disk will be encrypyted. Require `create_launch_template` to be `true` and require `disk_kms_key_id` to be set | bool | false |
+| disk\_kms\_key\_id | KMS Key used to encrypt the root disk. Require both `create_launch_template` and `disk_encrypted` to be `true` | string | "" |
```
Comment (johngmyers) [inline modules/node_groups/README.md:27]:
```suggestion
| disk\_encrypted | Whether the root disk will be encrypyted. Requires `create_launch_template` to be `true` and require `disk_kms_key_id` to be set | bool | false |
| disk\_kms\_key\_id | KMS Key used to encrypt the root disk. Requires both `create_launch_template` and `disk_encrypted` to be `true` | string | "" |
```

Comment (alzabo) [reply modules/node_groups/README.md:27]: I can certainly change this. I elected to use the singular as it was the existing style in the other README entries for the sake of consistency, though I agree it's a little stilted.

Comment (johngmyers) [reply modules/node_groups/README.md:27]: It's more than just stilted, it's grammatically incorrect.

Just noticed there's a second "require" on line 26 that also has incorrect number. It should probably be removed.

Comment (alzabo) [reply modules/node_groups/README.md]: Sorry for the delay. I updated the language in the README

**Review by bryantbiggs — APPROVED** [review]
@antonbabenko / @barryib 👍🏼

**Comment by jweigand** [comment]
Looks like this has one approval, any guess when it might be merged/released? Urgently needed for a project we're working on (which already uses the module).

**Comment by antonbabenko** [comment]
@jweigand Please see my other comment - https://github.com/terraform-aws-modules/terraform-aws-eks/pull/1459#issuecomment-901339428 . You will hear about this PR soon.

**Comment by antonbabenko** [comment]
Here we go! 🎉 

[v17.2.0](https://github.com/terraform-aws-modules/terraform-aws-eks/releases/v17.2.0) has been just released.

**Pull request merged**